File: Team04_Angular/src/app/Classes/escalation-request-dto.ts
export class EscalationRequestDto {
  escalationId: number;
  ticketId: number;
  previousEmployeeName: string;
  previousEmployeeSurname: string;
  reasonForEscalation: string;
  dateOfEscalation: Date;
  priorityName: string;
  tagName: string;
  ticketDescription: string;
  departmentId: number | null;

  constructor(
    escalationId: number,
    ticketId: number,
    previousEmployeeName: string,
    previousEmployeeSurname: string,
    reasonForEscalation: string,
    dateOfEscalation: Date,
    priorityName: string,
    tagName: string,
    ticketDescription: string,
    departmentId: number | null
  ) {
    this.escalationId = escalationId;
    this.ticketId = ticketId;
    this.previousEmployeeName = previousEmployeeName;
    this.previousEmployeeSurname = previousEmployeeSurname;
    this.reasonForEscalation = reasonForEscalation;
    this.dateOfEscalation = dateOfEscalation;
    this.priorityName = priorityName;
    this.tagName = tagName;
    this.ticketDescription = ticketDescription;
    this.departmentId = departmentId;
  }
}

File: Team04_API/Team04_API/Models/DTOs/EscalationRequestDto.cs
using System;

namespace Team04_API.Models.DTOs
{
    public class EscalationRequestDto
    {
        public int EscalationId { get; set; }
        public int TicketId { get; set; }
        public string? PreviousEmployeeName { get; set; }
        public string? PreviousEmployeeSurname { get; set; }
        public string? ReasonForEscalation { get; set; }
        public DateTime DateOfEscalation { get; set; }
        public string? PriorityName { get; set; }
        public string? TagName { get; set; }
        public string? TicketDescription { get; set; }
        public int? DepartmentId { get; set; } 
    }
}
